fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt

```diff
@@ -33,12 +33,23 @@ import org.mozilla.fenix.theme.FirefoxTheme
  *
  * @param tabsTrayStore [TabsTrayStore] used to listen for changes to [TabsTrayState].
  * @param displayTabsInGrid Whether the normal and private tabs should be displayed in a grid.
+ * @param onTabClose Invoked when the user clicks to close a tab.
+ * @param onTabMediaClick Invoked when the user interacts with a tab's media controls.
+ * @param onTabClick Invoked when the user clicks on a tab.
+ * @param onTabMultiSelectClick Invoked when the user clicks on a tab while in multi-select mode.
+ * @param onTabLongClick Invoked when the user long clicks a tab.
  */
 @OptIn(ExperimentalPagerApi::class, ExperimentalComposeUiApi::class)
+@Suppress("LongMethod", "LongParameterList")
 @Composable
 fun TabsTray(
     tabsTrayStore: TabsTrayStore,
     displayTabsInGrid: Boolean,
+    onTabClose: (TabSessionState) -> Unit,
+    onTabMediaClick: (TabSessionState) -> Unit,
+    onTabClick: (TabSessionState) -> Unit,
+    onTabMultiSelectClick: (TabSessionState) -> Unit,
```

Comment:
So validating this part is probably hard currently because the selection banner UI will be done [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1816522) and the selection logic (i.e. feeding the selection state to the UI) will be done [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1821425).

I'm able to enter multi selection with 1 tab, so not sure what's going on there. I think what you're seeing with the latter point is not a bug but unclear because the selection state has not been plumbed in yet. (and maybe that should have been done before this ticket).

Reply:
Yeah I can enter multiselect mode with no issues, but clicks while in multiselect mode seem a bit weird to me. Though I am happy to push that back to a more appropriate follow-up, just wanted to make sure you were aware

Reply:
Appreciate your due diligence here. I'll add a note to [1821425](https://bugzilla.mozilla.org/show_bug.cgi?id=1821425) to ensure behavior parity and bump-up that ticket in my priority list.